identity/client/src/pages/cluster-variables/List.tsx

```diff
@@ -0,0 +1,138 @@
+import DetailsModal from "src/pages/cluster-variables/modals/DetailsModal.tsx";
+
+export default function List() {
+  const { t } = useTranslate("clusterVariables");
+  const {
+    data: clusterVariables,
+    loading,
+    reload,
+    success,
+    search,
+    ...paginationProps
+  } = usePaginatedApi(searchClusterVariables);
+
+  const [addClusterVariable, addClusterVariableModal] = useModal(
+    AddModal,
+    reload,
+  );
+  const [deleteClusterVariable, deleteClusterVariableModal] = useEntityModal(
+    DeleteModal,
+    reload,
+  );
+  const [viewClusterVariable, viewClusterVariableModal] = useEntityModal(
+    DetailsModal,
+    () => {},
+  );
+
+  const shouldShowEmptyState =
+    success && !search && !clusterVariables?.items.length;
+
+  const pageHeader = (
+    <PageHeader
+      title={t("clusterVariables")}
+      linkText={t("clusterVariables").toLowerCase()}
+      docsLinkPath="/docs/components/identity/cluster-variable/"
+      shouldShowDocumentationLink={!shouldShowEmptyState}
+    />
+  );
+
+  if (shouldShowEmptyState) {
+    return (
+      <Page>
+        {pageHeader}
+        <PageEmptyState
+          resourceTypeTranslationKey={"clusterVariable"}
+          docsLinkPath="/docs/components/identity/cluster-variable/"
+          handleClick={addClusterVariable}
+        />
+        {addClusterVariableModal}
+      </Page>
+    );
+  }
+
+  return (
+    <Page>
+      {pageHeader}
+      <EntityList
+        data={
+          clusterVariables?.items.map((clusterVar) => {
+            return {
+              ...clusterVar,
+              value: clusterVar.value,
+              scopeValue:
+                clusterVar.scope === ScopeType.GLOBAL
+                  ? t("clusterVariableScopeTypeGlobal")
+                  : `${t("clusterVariableScopeTypeTenant")}: ${clusterVar.tenantId}`,
+            };
+          }) || []
+        }
+        headers={[
+          { header: t("name"), key: "name", isSortable: true },
+          { header: t("value"), key: "value" },
+          { header: t("scope"), key: "scopeValue" },
+        ]}
+        addEntityLabel={t("createClusterVariable")}
+        onAddEntity={addClusterVariable}
+        loading={loading}
+        menuItems={[
+          {
+            label: t("view"),
+            icon: View,
+            onClick: ({ name, value }) => {
+              viewClusterVariable({
+                name,
+                value,
+              });
+            },
+          },
+          {
+            label: t("delete"),
+            icon: TrashCan,
+            isDangerous: true,
+            onClick: ({ name, scope, tenantId }) => {
+              deleteClusterVariable({
+                name,
+                scope,
+                tenantId,
+              });
+            },
+          },
+        ]}
+        searchPlaceholder={t("searchByClusterVariableName")}
+        searchKey="name"
+        {...paginationProps}
+      />
+      {!loading && !success && (
```

Comment:
Shouldn't this be before/instead of the list rendering? Will the list rendering work without errors, if data is not available, because an error was returned?